WordPress/openverse
# Reduce ES shard count and simplify index properties

## Series

This PR is part 1/3.

1. This PR
2. #3161 
3. #3162

## Fixes



Fixes #2154 by @sarayourfriend.

Builds upon past work from WordPress/openverse-api#684 and WordPress/openverse-api#850.

## Description



This PR
- removes ES fields not used for filtering or searching
- converts text fields to keywords where partial matching is not needed
- updates usages of these changed fields in the API
- refactors ES models to DRY some common code
- reduces the shards for the audio index as a way to improve performance

## Testing Instructions


CI should pass. Compare index sizes (using something like Elasticvue) before and after the PR to see the difference.

- `just down -v`
- `just dc build web ingestion_server`
- `just api/init`

## Deployment

This PR needs a multi-step deployment strategy that is well-documented in #2154 and described in a [Google Doc](https://docs.google.com/document/d/13yg7Hpftw_emjVY08mNxfbvVAdll17oswAW5kx8Gjmw/edit?usp=sharing).

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
